app/services/requirements/scheduled_datetime_checker.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module Requirements
+  class ScheduledDatetimeChecker
+    attr_reader :day, :month, :year, :time
+
+    FUTURE_TIME_PERIOD = { months: 14 }.freeze
+
+    def initialize(params)
+      @day = is_numeric?(params[:day]) ? params[:day].to_i : params[:day]
+      @month = is_numeric?(params[:month]) ? params[:month].to_i : params[:month]
+      @year = is_numeric?(params[:year]) ? params[:year].to_i : params[:year]
+      @time = params[:time]
+    end
+
+    def date_issues
+      issues = []
+
+      begin
+        parsed_date
```

Comment:
I'm not sure if we're covering all the cases in these methods, but it might be the whacky behaviour of these classes: in `time_issues`, it looks like we're not expecting the parsing to blow up, while in the `date_issues`, it looks like we're not fussed if it's empty, but that sounds like it should be an issue. Have I missed something?

Reply:
I had expected Time.parse to work in the same way as Date.parse (i.e. throwing an ArgumentError) but turns out it just returns nil if you pass in an empty string or alphanumerical characters.